### PR TITLE
Python 2 to 3 updates - remove six

### PR DESCRIPTION
**Pull request by csabella**
This removes `six` from the code, but not the requirements or installation.

### CODE REVIEW

**Comment by JohnHillegass** [comment]
after removing all of these references, would we also want to remove the package?

**Comment by csabella** [comment]
> after removing all of these references, would we also want to remove the package?

Probably, but I thought it might be safer to do it in two steps.

**Comment by kapilt** [comment]
@JohnHillegass while its possible to remove setup.py its marginal value since six is a dep in our dependencies (ie remains in requirements/poetry.lock) and users need to install. at least for cloud sdks its unclear that they will be able to move off away from it for several years (think enterprise linux distros support).

**Comment by JohnHillegass** [comment]
Thanks for clarifying. I didn't know that before.

File: c7n/actions/network.py [hunk 1]
```diff
@@ -121,7 +120,7 @@ def _format_error(self, msg, **kw):
 
     def _get_array(self, k):
         v = self.data.get(k, [])
-        if isinstance(v, six.string_types):
+        if isinstance(v, str):
```
Comment (kapilt) [inline c7n/actions/network.py:123]: given the usage, it seems fine as (str, byte)

File: c7n/filters/core.py [hunk 2]
```diff
@@ -578,7 +577,7 @@ def match(self, i):
         return False
 
     def process_value_type(self, sentinel, value, resource):
-        if self.vtype == 'normalize' and isinstance(value, six.string_types):
+        if self.vtype == 'normalize' and isinstance(value, str):
```
Comment (kapilt) [inline c7n/filters/core.py:580]: this right hand side comparison on redundant and can just be removed.

File: tools/c7n_azure/c7n_azure/provisioning/deployment_unit.py [hunk 3]
```diff
@@ -1,14 +1,12 @@
...
 
 from abc import ABCMeta, abstractmethod
 
 from c7n.utils import local_session
 from c7n_azure.session import Session
 
 
-@six.add_metaclass(ABCMeta)
-class DeploymentUnit:
+class DeploymentUnit(metaclass=ABCMeta):
```
Comment (kapilt) [inline tools/c7n_azure/c7n_azure/provisioning/deployment_unit.py:9]: does this need a metaclass?

**Comment by csabella** [comment]
I should have made this comment earlier, but the way I converted from the `six` types to Python 3 was based on this code from `six`:
```
if PY3:
    string_types = str,
    integer_types = int,
    class_types = type,
    text_type = str
    binary_type = bytes
else:
    string_types = basestring,
    integer_types = (int, long)
    class_types = (type, types.ClassType)
    text_type = unicode
    binary_type = str
```

File: tools/c7n_azure/c7n_azure/provisioning/deployment_unit.py (hunk 3 above)
Comment (csabella) [reply tools/c7n_azure/c7n_azure/provisioning/deployment_unit.py:9]: In this case, I think it's used.  For example, `class AutoScaleUnit` in `autoscale.py` inherits from `DeploymentUnit` and is forced to implement `_get` and `_provision`.  There are other classes that are similar.

Comment (kapilt) [reply tools/c7n_azure/c7n_azure/provisioning/deployment_unit.py:9]: fair enough, thanks for checking

**Pull request merged**